File: src/test/generate_test_data/generateRunSaveTest_EMGLLF.m
function[] = generateRunSaveTest_EMGLLF(n, p, m, k, mini, maxi, gamma, lambda, varargin)

	%set defaults for optional inputs
	optargs = {200 15 10 3 5 10 1.0 0.5};
	%replace defaults by user parameters
	optargs(1:length(varargin)) = varargin;
	[n, p, m, k, mini, maxi, gamma, lambda] = optargs{:};
	tau = 1e-6;
	mini = int64(mini);
	maxi = int64(maxi);

	%Generate phiInit,piInit,...
	[phiInit,rhoInit,piInit,gamInit] = basicInitParameters(n, p, m, k);

	%Generate X and Y
	[X, Y, ~] = generateIOdefault(n, p, m, k);

	%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

	testFolder = 'data/';
	mkdir(testFolder);
	delimiter = ' ';

	%save inputs
	dlmwrite(strcat(testFolder,'phiInit'), reshape(phiInit,1,[]), delimiter);
	dlmwrite(strcat(testFolder,'rhoInit'), reshape(rhoInit,1,[]), delimiter);
	dlmwrite(strcat(testFolder,'piInit'), piInit, delimiter);
	dlmwrite(strcat(testFolder,'gamInit'), reshape(gamInit,1,[]), delimiter);
	dlmwrite(strcat(testFolder,'mini'), mini, delimiter);
	dlmwrite(strcat(testFolder,'maxi'), maxi, delimiter);
	dlmwrite(strcat(testFolder,'gamma'), gamma, delimiter);
	dlmwrite(strcat(testFolder,'lambda'), lambda, delimiter);
	dlmwrite(strcat(testFolder,'X'), reshape(X,1,[]), delimiter);
	dlmwrite(strcat(testFolder,'Y'), reshape(Y,1,[]), delimiter);
	dlmwrite(strcat(testFolder,'tau'), tau, delimiter);
	dlmwrite(strcat(testFolder,'dimensions'), [n,p,m,k], delimiter);

	[phi,rho,pi,LLF,S] = EMGLLF(phiInit,rhoInit,piInit,gamInit,mini,maxi,gamma,lambda,X,Y,tau);

	%save output
	dlmwrite(strcat(testFolder,'phi'), reshape(phi,1,[]), delimiter);
	dlmwrite(strcat(testFolder,'rho'), reshape(rho,1,[]), delimiter);
	dlmwrite(strcat(testFolder,'pi'), pi, delimiter);
	dlmwrite(strcat(testFolder,'LLF'), LLF, delimiter);
	dlmwrite(strcat(testFolder,'S'), reshape(S,1,[]), delimiter);

end

File: src/test/generate_test_data/generateRunSaveTest_EMGLLF.R
generateRunSaveTest_EMGLLF = function(n=200, p=15, m=10, k=3, mini=5, maxi=10,
	gamma=1., lambda=0.5, tau=1e-6)
{
	testFolder = "data/"
	dir.create(testFolder, showWarnings=FALSE, mode="0755")

	require(valse)
	params = valse:::basic_Init_Parameters(n, p, m, k)
	io = generateIOdefault(n, p, m, k)

	#save inputs
	write.table(as.double(params$phiInit), paste(testFolder,"phiInit",sep=""),
		row.names=F, col.names=F)
	write.table(as.double(params$rhoInit), paste(testFolder,"rhoInit",sep=""),
		row.names=F, col.names=F)
	write.table(as.double(params$piInit), paste(testFolder,"piInit",sep=""),
		row.names=F, col.names=F)
	write.table(as.double(params$gamInit), paste(testFolder,"gamInit",sep=""),
		row.names=F, col.names=F)
	write.table(as.integer(mini), paste(testFolder,"mini",sep=""),
		row.names=F, col.names=F)
	write.table(as.integer(maxi), paste(testFolder,"maxi",sep=""),
		row.names=F, col.names=F)
	write.table(as.double(gamma), paste(testFolder,"gamma",sep=""),
		row.names=F, col.names=F)
	write.table(as.double(lambda), paste(testFolder,"lambda",sep=""),
		row.names=F, col.names=F)
	write.table(as.double(io$X), paste(testFolder,"X",sep=""),
		row.names=F, col.names=F)
	write.table(as.double(io$Y), paste(testFolder,"Y",sep=""),
		row.names=F, col.names=F)
	write.table(as.double(tau), paste(testFolder,"tau",sep=""),
		row.names=F, col.names=F)
	write.table(as.integer(c(n,p,m,k)), paste(testFolder,"dimensions",sep=""),
		row.names=F, col.names=F)

	res = EMGLLF(params$phiInit,params$rhoInit,params$piInit,params$gamInit,mini,maxi,
		gamma,lambda,io$X,io$Y,tau)

	#save outputs
	write.table(as.double(res$phi), paste(testFolder,"phi",sep=""), row.names=F, col.names=F)
	write.table(as.double(res$rho), paste(testFolder,"rho",sep=""), row.names=F, col.names=F)
	write.table(as.double(res$pi), paste(testFolder,"pi",sep=""), row.names=F, col.names=F)
	write.table(as.double(res$LLF), paste(testFolder,"LLF",sep=""), row.names=F, col.names=F)
	write.table(as.double(res$S), paste(testFolder,"S",sep=""), row.names=F, col.names=F)
}
